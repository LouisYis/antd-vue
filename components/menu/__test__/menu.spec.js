import Vue from 'Vue'
import { Menu, MenuItem, SubMenu } from '../index'

let selectId = null

const MenuInstance = new Vue({
  components: {
    [Menu.name]: Menu,
    [MenuItem.name]: MenuItem,
    SubMenu
  },
  methods: {
    getChild(id, len) {
      let children = []
      for (let i = 0; i < len; i++) {
        children.push(<vk-menu-item keyVal={`${id}.${i}`}>{`${id}.${i}`}</vk-menu-item>)
      }
      return children
    }
  },
  render(h) {
    const props = {
      type: 'vertical',
      defaultSelect: '1.0',
      autoClose: true,
      onSelect(key) {
        selectId = key
      }
    }
    return (
      <vk-menu {...{props}}>
        <SubMenu title="item 1">
          {this.getChild(1, 3)}
          <SubMenu title="item 5">
            {this.getChild(2, 3)}
          </SubMenu>
        </SubMenu>
        <vk-menu-item key-val="2">item 2</vk-menu-item>
        <SubMenu title="item 3">
          {this.getChild(3, 2)}
        </SubMenu>
        <vk-menu-item key-val="4">item 4</vk-menu-item>
      </vk-menu>
    )
  }
}).$mount()

describe('Menu component', () => {
  test('component should match snapshot', () => {
    expect(MenuInstance.$el).toMatchSnapshot()
  })

  test('component select func run currectly', () => {
    expect(selectId).not.toBeNull
  })

  test('component props required', () => {
    expect(Menu.props.onSelect).not.toBeUndefined()
  })
})

describe('Menu item component', () => {
  test('component props required', () => {
    expect(MenuItem.props.keyVal).not.toBeUndefined()
  })
})

describe('SubMenu item component', () => {
  test('component props required', () => {
    expect(SubMenu.props.title).not.toBe(undefined)
  })

  test('component data required', () => {
    expect(SubMenu.data().isOpen).toBe(false)
  })
})
