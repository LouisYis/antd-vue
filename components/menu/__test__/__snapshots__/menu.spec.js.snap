// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component component should match snapshot 1`] = `
<ul
  class="vk-menu is-vertical"
>
  <li
    class="vk-sub-menu active"
  >
    <div
      class="vk-sub-menu-title"
    >
      item 1
    </div>
    <ul
      class="vk-menu is-float"
      style="display: none;"
    >
      <li
        class="vk-menu-item active"
      >
        1.0
      </li>
      <li
        class="vk-menu-item"
      >
        1.1
      </li>
      <li
        class="vk-menu-item"
      >
        1.2
      </li>
      <li
        class="vk-sub-menu"
      >
        <div
          class="vk-sub-menu-title"
        >
          item 5
        </div>
        <ul
          class="vk-menu is-float"
          style="display: none;"
        >
          <li
            class="vk-menu-item"
          >
            2.0
          </li>
          <li
            class="vk-menu-item"
          >
            2.1
          </li>
          <li
            class="vk-menu-item"
          >
            2.2
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    class="vk-menu-item"
  >
    item 2
  </li>
  <li
    class="vk-sub-menu"
  >
    <div
      class="vk-sub-menu-title"
    >
      item 3
    </div>
    <ul
      class="vk-menu is-float"
      style="display: none;"
    >
      <li
        class="vk-menu-item"
      >
        3.0
      </li>
      <li
        class="vk-menu-item"
      >
        3.1
      </li>
    </ul>
  </li>
  <li
    class="vk-menu-item"
  >
    item 4
  </li>
</ul>
`;
